runtime/src/impls.rs

```diff
@@ -30,89 +31,102 @@ impl Convert<u128, Balance> for CurrencyToVoteHandler {
-	fn convert(multiplier: Multiplier) -> Multiplier {
-		let max_weight = MaximumBlockWeight::get();
-		let block_weight = System::block_weight().total().min(max_weight);
-		let target_weight = (T::get() * max_weight) as u128;
-		let block_weight = block_weight as u128;
-
-		// determines if the first_term is positive
-		let positive = block_weight >= target_weight;
-		let diff_abs = block_weight.max(target_weight) - block_weight.min(target_weight);
-		// safe, diff_abs cannot exceed u64.
-		let diff = Multiplier::saturating_from_rational(diff_abs, max_weight.max(1));
-		let diff_squared = diff.saturating_mul(diff);
-
-		// 0.00004 = 4/100_000 = 40_000/10^9
-		let v = Multiplier::saturating_from_rational(4, 100_000);
-		// 0.00004^2 = 16/10^10 Taking the future /2 into account... 8/10^10
-		let v_squared_2 = Multiplier::saturating_from_rational(8, 10_000_000_000u64);
-
-		let first_term = v.saturating_mul(diff);
-		let second_term = v_squared_2.saturating_mul(diff_squared);
-
-		if positive {
-			// Note: this is merely bounded by how big the multiplier and the inner value can go,
-			// not by any economical reasoning.
-			let excess = first_term.saturating_add(second_term);
-			multiplier.saturating_add(excess)
-		} else {
-			// Defensive-only: first_term > second_term. Safe subtraction.
-			let negative = first_term.saturating_sub(second_term);
-			multiplier.saturating_sub(negative)
-				// despite the fact that apply_to saturates weight (final fee cannot go below 0)
-				// it is crucially important to stop here and don't further reduce the weight fee
-				// multiplier. While at -1, it means that the network is so un-congested that all
-				// transactions have no weight fee. We stop here and only increase if the network
-				// became more busy.
-				.max(Multiplier::saturating_from_integer(-1))
-		}
+/// Yet, it can be used for any other sort of change to weight-fee. Some examples being:
+///   - Setting it to `0` will essentially disable the weight fee.
+///   - Setting it to `1` will cause the literal `#[weight = x]` values to be charged.
+///
+/// Sample weight to Fee Calculation for 1 Rad Balance transfer:
+/// 	ExtrinsicBytes: 92 bytes
+/// 	Weight: 195000000
+/// 	WeightCoefficient: 315000
+/// 	TransactionByteFee: 0.01 Micro RAD = 10000000000
+///		MaximumBlockWeight: 2 * WEIGHT_PER_SECOND = 2000000000000
+/// 	ExtrinsicBaseWeight: 125 * WEIGHT_PER_MICROS = 125000000
+///
+/// 	Calculation:
+/// 	BaseFee = ExtrinsicBaseWeight * WeightCoefficient = 39375000000000
+/// 	LengthFee = ExtrinsicBytes * TransactionByteFee = 920000000000
+/// 	WeightFee = Weight * WeightCoefficient = 61425000000000
+/// 	Fee = BaseFee + LengthFee + WeightFee = 101720000000000 => 101.72 Micro RAD
+pub struct WeightToFee;
+impl WeightToFeePolynomial for WeightToFee {
+	type Balance = Balance;
+
+	fn polynomial() -> WeightToFeeCoefficients<Self::Balance> {
+		smallvec!(WeightToFeeCoefficient {
+			coeff_integer: 315000,
```

Comment:
I noticed that you came to a different weight to fee constant (315000) from my calculation (625000) – are you sure your calculation is correct? I haven’t looked into it yet, just want to make sure that our fees will not suddenly be much cheaper than before.

Reply:
Do verify this locally to ensure the fees are not lowered or made much cheaper than we are charging right now in mainnet @philipstanislaus

Reply:
@vedhavyas done, I confirmed locally that anchor commits and balance transfers stay roughly the same. Thanks!